src/features/authentication/register/register.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useEffect, useState } from "react";
+import {
+  createUserWithEmailAndPassword,
+  onAuthStateChanged,
+} from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+import { InputText } from "primereact/inputtext";
+import { Button } from "primereact/button";
+
+import logo from "../../../common/assets/Logo.png";
+import "./register.scss";
+
+import { auth } from "../../../config/firebase-config";
+import { observer } from "mobx-react-lite";
+
+interface RegisterFormProps {}
+
+interface UserData {}
+
+export const RegisterForm: React.FC<RegisterFormProps> = observer(() => {
+  const navigate = useNavigate();
+
+  const [user, setUser] = useState<UserData | null>(null);
+
+  const initialValues = {
+    email: "",
+    password: "",
+  };
+  const [formValues, setFormValues] = useState(initialValues);
+  const [formErrors, setFormErrors] = useState(initialValues);
+  const [isSubmit, setIsSubmit] = useState(false);
+
+  const register = async () => {
+    const errors = initialValues;
+    try {
+      const user = await createUserWithEmailAndPassword(
+        auth,
+        formValues.email,
+        formValues.password
+      );
+      navigate("/homepage");
+    } catch (error: any) {
+      if (error.message.includes("email-already-in-use")) {
+        errors.email = "Email already in use.";
+        setFormErrors(errors);
+      }
+      console.log(error.message);
+    }
+  };
+
+  const handleChange = (e: any) => {
+    const { name, value } = e.target;
+    setFormValues({ ...formValues, [name]: value });
+  };
+
+  const handleSubmit = (e: any) => {
+    e.preventDefault();
+    setFormErrors(validate(formValues));
+    setIsSubmit(true);
+  };
+
+  const validate = (values: any) => {
```

Comment:
I notice that this function is used in multiple places, so it could be moved into a common/services folder file and be taken anywhere it is needed.